eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -21,6 +21,10 @@ parameters:
 - name: architecture
   type: string
 
+- name: artifactsRid
+  type: string
+  default: centos.8-x64
```

Comment:
If you defaulted this to empty, you wouldn't be redefining the default value here.  If it's empty, don't pass in the `artifacts-rid` parameter to the prep script and let it define what the default should be.  This will make it easy to update the default in the future because it will only be defined in a single place within the code.

Reply:
https://github.com/dotnet/installer/pull/17370/files/bf697aa5cab242f80871e0db905b3d85ba24dedf#diff-dc549eeb99aa73960e74c9d4c160a3c76e3e414398bb348c1d5365aac0369ad5R157 requires that the artifactsRid be defined. What about defining the default as a variable in the pipeline, then letting the default parameter be empty as you suggested?

Reply:
I'm sorry I missed that usage.  What about defaulting the parameter to null. When the withPreviousSDK parameter is set the parameter would be required.  This makes sense to me because those legs are for a specific RID - the default rid for the prep just happens to align but that doesn't mean the default should be used.  When the OS is upgraded for those legs the rid would obviously also need updating.